scipy/stats/tests/test_stats.py

```diff
@@ -3230,112 +3234,163 @@ def _assert_equal(self, actual, expect, *, shape=None, dtype=None):
             dtype = expect.dtype
         assert actual.dtype == dtype
 
+    @array_api_compatible
     @pytest.mark.parametrize('size', [10, (10, 2)])
     @pytest.mark.parametrize('m, c', product((0, 1, 2, 3), (None, 0, 1)))
-    def test_moment_center_scalar_moment(self, size, m, c):
+    def test_moment_center_scalar_moment(self, size, m, c, xp):
         rng = np.random.default_rng(6581432544381372042)
-        x = rng.random(size=size)
+        x = xp.asarray(rng.random(size=size))
         res = stats.moment(x, m, center=c)
-        c = np.mean(x, axis=0) if c is None else c
-        ref = np.sum((x - c)**m, axis=0)/len(x)
-        assert_allclose(res, ref, atol=1e-16)
+        c = xp.mean(x, axis=0) if c is None else c
+        ref = xp.sum((x - c)**m, axis=0)/x.shape[0]
+        xp_assert_close(res, ref, atol=1e-16)
 
+    @array_api_compatible
     @pytest.mark.parametrize('size', [10, (10, 2)])
     @pytest.mark.parametrize('c', (None, 0, 1))
-    def test_moment_center_array_moment(self, size, c):
+    def test_moment_center_array_moment(self, size, c, xp):
         rng = np.random.default_rng(1706828300224046506)
-        x = rng.random(size=size)
+        x = xp.asarray(rng.random(size=size))
         m = [0, 1, 2, 3]
         res = stats.moment(x, m, center=c)
-        ref = [stats.moment(x, i, center=c) for i in m]
-        assert_equal(res, ref)
+        xp = array_namespace(x)
+        ref = xp.concat([stats.moment(x, i, center=c)[xp.newaxis, ...] for i in m])
```

Comment:
Just to help mark that verbosity can be reduced once our minimum NumPy version is 2.0.

```suggestion
        xp_test = array_namespace(x) # np<2.0 does not have `xp.concat`
        ref = xp_test.concat([stats.moment(x, i, center=c)[xp.newaxis, ...] for i in m])
```